tools/influx_si/influx_si.xml

```diff
@@ -1,232 +1,74 @@
-            #end if
-            $opt.nocalc
-            $opt.addnoise
-            $opt.TIMEIT
-            #if $opt.tblimit:
-             --tblimit='$opt.tblimit'
+<tool id="influx_si" name="influx_si" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="21.09">
+    <description>
+        Estimate metabolic fluxes and concentrations by fitting simulated labeling in metabolites to NMR/MS measurements
+    </description>
+    <creator>
+        <person givenName="Serguei" familyName="Sokol" email="sokol@insa-toulouse.fr" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/cellule-mathematiques/" />
+        <organization name="INRAE" url="https://www.inrae.fr/" />
+        <organization name="TBI" url="https://www.toulouse-biotechnology-institute.fr/" />
+        <organization name="Mathematics Cell" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/cellule-mathematiques/" />
+        <organization name="MetaToul-FluxoMet" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/metatoul/" />
+        <organization name="MetaboHub2" url="https://www.metabohub.fr/" />
+    </creator>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">influx_si</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir outdir;
+        #for $inp in $input_main:
+            unzip $inp;
+        #end for
+               
+        ret=0;
+        influx_$si.s_i
+            @COMMAND_OPT@
+            #if $si.s_i == 'i' and $si.time_order != 'None':
+              --time_order='$si.time_order'
             #end if
+            #for $inp in $input_main:
+                #set base = $inp.name[:-4]
+                --prefix "$base"
+            #end for
+            @MTF_OPT@
+            > 'outdir/influx_${si.s_i}.stdout.txt' 2>'outdir/influx_${si.s_i}.stderr.txt' ||
+            ret=\${PIPESTATUS[0]};
+
 
-         #if $si.s_i == 'i' and $si.time_order:
-           --time_order='$si.time_order'
-         #end if
-         '${"' '".join([str($in_m.element_identifier) for $in_m in $input_main])}'
-         > 'influx_${si.s_i}.stdout.txt' 2>'influx_${si.s_i}.stderr.txt'; res="\$?" &&
-       
-       for f in *.ftbl *.log *.err *.R *.kvh *.attrs; do mv -f "\$f" "\$f".txt || true; done &&
-       
-       for f in *.pdf; do [ -e "\$f" ] && mv -f "\$f" "\$f".pdf || true; done &&
-       
-       for f in *.RData; do [ -e "\$f" ] && mv -f "\$f" "\$f".RData || true; done &&
-       [ "\$res" == "0" ]
+        #for $inp in $input_main:
+            #set base = $inp.name[:-4]
+            for f in "$base"_res/*.{err,log}; do mv -f "\$f" "\$f".txt || true; done;
+            for f in "$base"_res/*.{sim,stat}; do mv -f "\$f" "\$f".tsv || true; done;
+            for f in "$base"_res/*.pdf; do [ -e "\$f" ] && mv -f "\$f" "\$f".pdf || true; done;
```

Comment:
Well caught for existence check but adding extra ".pdf" is a workaround against galaxy's "feature" to keep only the name (without extension) of a file in a collection. In my result collection, I have many files with the same name and different extensions, e.g. ecoli.log, ecoli.err, ecoli.pdf, etc. If I leave 'ecoli.pdf' as is, it will appear as 'ecoli' without giving an immediate hint on what type it is. Moreover, .pdf file may or may be not among the result files, depending of user's request. So appearing or disappearing it just as 'ecoli' may be confusing.

Reply:
This would all be covered by my suggested changes to the collection element discovery.